src/graph.rs

```diff
@@ -2296,21 +2891,116 @@ impl<'a, 'graph> Builder<'a, 'graph> {
       }
     }
 
+    let specifier = specifier.clone();
+    self.load_pending_module(&specifier, maybe_range, &specifier, is_dynamic);
+  }
+
+  fn load_pending_module(
+    &mut self,
+    requested_specifier: &Url,
+    maybe_range: Option<Range>,
+    load_specifier: &Url,
+    is_dynamic: bool,
+  ) {
     self
       .graph
       .module_slots
-      .insert(specifier.clone(), ModuleSlot::Pending);
-    let specifier = specifier.clone();
+      .insert(requested_specifier.clone(), ModuleSlot::Pending);
+    let load_specifier = load_specifier.clone();
+    let requested_specifier = requested_specifier.clone();
     let fut =
       self
         .loader
-        .load(&specifier, is_dynamic)
+        .load(&load_specifier, is_dynamic)
         .map(move |result| PendingInfo {
-          specifier,
+          specifier: requested_specifier,
           maybe_range,
-          result,
+          result: result.map(|r| r.map(Into::into)),
+          maybe_version_info: None,
         });
-    self.pending.push_back(Box::pin(fut));
+    self.state.pending.push_back(Box::pin(fut));
+  }
+
+  fn queue_load_package_info(&mut self, package_name: &str) {
+    if self
+      .state
+      .deno
+      .pending_package_info_loads
+      .contains_key(package_name)
+    {
+      return; // already queued
+    }
+
+    // request to load
+    let specifier = self
+      .loader
+      .registry_url()
+      .join(&format!("{}/meta.json", package_name))
+      .unwrap();
```

Comment:
Just a thought - maybe abstract away into helper functions to obtains meta URLs for packages and versions? It appears it's happening in a few places right now